### PR TITLE
[CSL-806] Investigate duplicate events - Use indexeddb

### PR DESCRIPTION
**Pull request by esezen**
Alternative solution. Uses indexeddb instead of localStorage. Adds less than 1kb to bundle size

### CODE REVIEW

File: src/utils/request-queue.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable brace-style, no-unneeded-ternary */
 const fetchPonyfill = require('fetch-ponyfill');
 const Promise = require('es6-promise');
-const store = require('../utils/store');
+const idb = require('idb-keyval/dist/compat');
```
Comment (esezen) [inline src/utils/request-queue.js:4]: IE10 compatible

File: spec/mocha.helpers.js [hunk 2]
```diff
@@ -10,6 +12,9 @@ const setupDOM = () => {
   global.window = window;
   global.document = window.document;
   global.AbortController = window.AbortController;
+  global.navigator = { indexedDB };
+  global.indexedDB = indexedDB;
+  idb.clear();
```
Comment (esezen) [inline spec/mocha.helpers.js:17]: This needs to change

File: src/utils/request-queue.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable brace-style, no-unneeded-ternary */
 const fetchPonyfill = require('fetch-ponyfill');
 const Promise = require('es6-promise');
-const idb = require('idb-keyval/dist/compat');
+const idb = require('idb-keyval');
```
Comment (esezen) [inline src/utils/request-queue.js:4]: We should figure out how to make tests work with compat version

**Comment by sblaurock** [comment]
Closing as switching to synchronous data storage pattern has large implementation implications.

**Pull request closed without merging**